VisualizationBase/src/views/MainView.cpp

```diff
@@ -111,6 +115,29 @@ void MainView::wheelEvent(QWheelEvent *event)
 	// Zoom
 	if (event->modifiers() == Qt::ControlModifier)
 	{
+		QPointF eventToViewportMiddleOffset;
+		QGraphicsItem* itemUnderCursor = nullptr;
+		QRectF boundingRectBeforeZoom;
+		QPointF eventPosInItem;
+
+		if (ITEM_STRUCTURE_AWARE_ZOOM_ANCHORING)
+		{
+			// find offset of event pos from middle of viewport
+			eventToViewportMiddleOffset = event->posF() - QPointF(viewport()->width()/2.0, viewport()->height()/2.0);
+			auto scenePos = mapToScene(event->pos());
+			auto itemsAtEvent = items(event->pos());
+			auto iter = itemsAtEvent.begin();
+			// avoid zooming in on Overlays
+			// TODO cast *iter to Item to use DCast or is this fine?
```

Comment:
In general, for things that inherit from `Item` or `Node`, always use `DCast`. It's faster.